Equilobe.DailyReport.SL/ReportService.cs

```diff
@@ -0,0 +1,114 @@
+﻿using Equilobe.DailyReport.DAL;
+using Equilobe.DailyReport.Models.Interfaces;
+using Equilobe.DailyReport.Models.Jira;
+using Equilobe.DailyReport.Models.ReportFrame;
+using Equilobe.DailyReport.Models.Storage;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Equilobe.DailyReport.SL
+{
+    public class ReportService : IReportService
+    {
+        public IJiraService JiraService { get; set; }
+
+        #region IJiraService Implementation
+        public void UpdateDashboardData(long instanceId)
+        {
+            var projects = GetProjectsFromInstance(instanceId);
```

Comment:
We won't use this. If we get the users per instance we do not to get them from their projects and then add them. Just get directly the users from one instance.

Reply:
There is no jira api request to get all the users from the instance. There is a workaround by making a search through multiple projects, but the projects keys are needed, so we must make a request to get the projects keys. Using the search could mean only one request, not making a request per project, which would be faster, but i would need to make a request to get projects keys.

I will try it now

Reply:
Actually, I now have looked into id, and the method searching in all projects is meant for getting the users who are members in all projects, not get all of the member from all the projects. So there is no other way than to get manually, from each project, the members

Reply:
Hmm. Ok. I see that there are some ways to get the users, but they do not seem reliable. One of them would be to use ' /search?username=%' but I've seen that it doesn't work in all Jira versions, so it doesn't seem that reliable.

Reply:
https://community.atlassian.com/t5/Jira-questions/Jira-rest-api-get-all-users/qaq-p/545237
Can this help you?

Reply:
I've tried this. But I think I might get a way to get all the users with one call.

Reply:
@rungureanu-eq That is the same thing that I suggested, but from what I read, this changed from version to version. So it is not very reliable.